src/segment_strava.go

```diff
@@ -0,0 +1,258 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"math"
+	"net/http"
+	"time"
+)
+
+// segment struct, makes templating easier
+type strava struct {
+	props Properties
+	env   Environment
+
+	StravaData
+	ActivityIcon string
+	Ago          string
```

Comment:
We should use a template function for this. My plan was to extend the template functions to have these more generic, for example, the executiontime segment already has quite some "formatters" which can also be used elsewhere (I still need to do that though).

Reply:
@JanDeDobbeleer I think we're ready.